packages/framework/src/Model/Advert/Advert.php

```diff
@@ -70,6 +72,13 @@ class Advert
      */
     protected $hidden;
 
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Category\Category[]|\Doctrine\Common\Collections\Collection
+     * @ORM\ManyToMany(targetEntity="Shopsys\FrameworkBundle\Model\Category\Category")
+     * @ORM\JoinTable(name="advert_categories")
+     */
+    protected $categories;
```

Comment:
why do not we use strict typehints? (even in `AdvertData, there is a new property without the typehint) :innocent: with the suggestion above, you can add `Collection` typehint here.

Reply:
I have added typehints as you suggested